### ISSUE TITLE
Blur disappears when navigating away from and back to view in 1.6.3

### ISSUE THREAD

**Issue by danielwilson1702** [Dimezis/BlurView#117]
Please include:
1) Library version: 1.6.3. Issue _not_ present in 1.6.2.
2) Device and OS version: Android Oreo
3) Detailed steps to reproduce the issue: Create a BlurView, navigate away from the screen and back to it again, blur disappears.

Hi, this lib is still unmatched imo so thank you. I just thought you'd like to know, I have a very basic implementation of a BlurView that mistakenly clears itself if you navigate away from the screen with the view and come back to it. The issue is **not** there in 1.6.2 and below:

```
binding.welcomeTitleBlurView.setupWith(binding.welcomeBlurContainer)
                .setBlurAlgorithm(RenderScriptBlur(activity))
                .setBlurRadius(2F)
                .setHasFixedTransformationMatrix(true)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            binding.welcomeTitleBlurView.outlineProvider = ViewOutlineProvider.BACKGROUND
            binding.welcomeTitleBlurView.clipToOutline = true
        }
```

I am blurring in a Fragment's onCreateView. I noticed if I add a delay of 100ms it successfully blurs when it navigates back, and just in case I made sure to try and perform the blur after it has been laid out using:

```
    inline fun View.afterMeasured(crossinline f: View.() -> Unit) {
        viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
            override fun onGlobalLayout() {
                if (measuredWidth > 0 && measuredHeight > 0) {
                    viewTreeObserver.removeOnGlobalLayoutListener(this)
                    f()
                }
            }
        })
    }
```

So something has changed regarding lifecycle callbacks, or threading or something, not sure which.

Also, I noticed the blurring algorithm seems to stretch or magnify the blurred image differently in the setup I have in 1.6.2 / 1.6.3, whereas from 1.5.0->1.6.1 it is different. My view happens to be a relatively narrow, long view on a 7 inch landscape screen. I think 1.6.1 looks more accurate and doesn't exhibit the disappearing blur behavior in 1.6.3, so will stick with that for the moment.

Here is 1.6.1 with a blur radius of 2:
![image](https://user-images.githubusercontent.com/5859421/86615162-fe00ce80-bfab-11ea-9ad8-54dc622fb852.png)

Here is 1.6.2 with a blur radius of 2:
![image](https://user-images.githubusercontent.com/5859421/86615463-6485ec80-bfac-11ea-8cce-cac23e7c2553.png)

1.6.2 with an increased blur radius of 8 is better:
![image](https://user-images.githubusercontent.com/5859421/86615719-bf1f4880-bfac-11ea-9b0a-514f23188e1c.png)

**Comment by Dimezis** [comment]
Thank you for the detailed report.
As for the disappearing blur, I guess it's related to this https://github.com/Dimezis/BlurView/issues/89.
There was a small change, posting of some callbacks was removed, and seems like it introduced this bug instead.
That's why your postponing workaround seems to fix it.

Regarding the stretching, I believe it's related to a change in https://github.com/Dimezis/BlurView/releases/tag/version-1.6.2, I've increased the rounding value to 64 to significantly improve the performance on Samsung devices.
But apparently I forgot to scale the Y dimension accordingly.

Both should be fixable, but can't say for sure when I'll be able to do it

**Comment by danielwilson1702** [comment]
Nice one thank you yes it took me a little while to figure out the issue was the version change. I think 1.6.1 will be fine for me for the moment and will keep an eye out for any updates if you or anyone else has time so thanks!

**Comment by Dimezis** [comment]
@danielwilson1702 hey, could you please check if these issues are fixed for you on this branch?
https://github.com/Dimezis/BlurView/tree/scaling_refactoring

**Comment by Dimezis** [comment]
I released 1.6.4 with a fix, please check if it works for you

**Comment by danielwilson1702** [comment]
Hi @Dimezis sorry for the delay, I tried 1.6.4 and now no blur 😕

**Comment by Dimezis** [comment]
@danielwilson1702 by the way, does the blur appear if you manually call `setBlurAutoUpdate(true)` during the setup?

**Comment by danielwilson1702** [comment]
Sorry I forget to come back to you,  `setBlurAutoUpdate(true)` does indeed fix 1.6.4, so I guess it's not a problem any longer then?

**Comment by Dimezis** [comment]
Honestly, I got too lazy to get it fixed without any hacks, so if that's working for you, let it stay for now :)
